File: backend/src/main/java/com/back/ourlog/domain/comment/dto/CommentResponseDto.java
package com.back.ourlog.domain.comment.dto;

import com.back.ourlog.domain.comment.entity.Comment;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class CommentResponseDto {
    private int id;
    private Integer userId;
    private String nickname;
    private String profileImageUrl;
    private String content;
    private LocalDateTime createdAt;

    public CommentResponseDto(Comment comment) {
        id = comment.getId();
        userId = comment.getUser() != null ? comment.getUser().getId() : null;
        nickname = comment.getUser() != null ? comment.getUser().getNickname() : null;
        profileImageUrl = comment.getUser() != null ? comment.getUser().getProfileImageUrl() : null;
        content = comment.getContent();
        createdAt = comment.getCreatedAt();
    }
}

File: backend/src/main/java/com/back/ourlog/domain/comment/dto/CommentResponseDto.kt
package com.back.ourlog.domain.comment.dto

import com.back.ourlog.domain.comment.entity.Comment
import java.time.LocalDateTime

data class CommentResponseDto(
    val id: Int,
    val userId: Int,
    val nickname: String,
    val profileImageUrl: String ?,
    val content: String,
    val createdAt: LocalDateTime
) {
    constructor(comment: Comment) : this (
        id = comment.id,
        userId = comment.user.id ?: throw IllegalStateException("Comment user ID cannot be null"),
        nickname = comment.user.nickname,
        profileImageUrl = comment.user.profileImageUrl,
        content = comment.content,
        createdAt = comment.createdAt
    )
}